tests/test_prereleases.py

```
@@ -0,0 +1,927 @@
+    # Verify old directories exist
+    assert old_dir1.exists()
+    assert old_dir2.exists()
+
+    # Mock the repo to return a newer prerelease with same version but new hash
+    with patch("fetchtastic.downloader.menu_repo.fetch_repo_directories") as mock_dirs:
+        with patch(
+            "fetchtastic.downloader.menu_repo.fetch_directory_contents"
+        ) as mock_contents:
+            mock_dirs.return_value = ["firmware-2.7.7.789abc"]
+
+            def _dir_aware_contents(dir_name: str):
```

Ruff (0.14.2)

Unused function argument: `dir_name`

(ARG001)